### PR TITLE
Replace pep8 with pycodestyle

### PR DESCRIPTION
**Pull request by newellz2**
The pep8 tool changed its name to pycodestyle a few years ago and added more errors. This commit allows one to run 'make pycodestyle' instead of 'make pep8'. 'make pycodestyle' produces many style related errors. Errors will have to be added to PEP8_IGNORE or python source code will have to be altered. I used the 2.6.0 version of pycodestyle that is included in Debian bullseye to test everything. I added E114, which is relevant to comment whitespace, to the PEP8_IGNORE variable. It eliminates\hides 2,540 errors.

I also added a 'make pycodestyle-stats' target to print out code style statistics.

Signed-off-by: Zachary Newell <newellz2@unr.edu>

### CODE REVIEW

**Review by saschalucas — APPROVED** [review]
Thanks @newellz2, looks good!

Would you be so kind and also replace/cleanup pep8 occurrence in `doc/devnotes.rst`? I know this file is horribly outdated, but hopefully it will converge to a better state.

I'll wait for your reply before merging ...

**Comment by newellz2** [comment]
@saschalucas, I cleaned up `doc/devnotes.rst` and pushed the commit. I need to find some time to figure out the issues with the Haskell portion.

**Review by saschalucas — APPROVED** [review]
Thanks @newellz2 for reworking the devnotes.

**Pull request merged**
